src/stacked_borrows/diagnostics.rs

```diff
@@ -331,6 +327,17 @@ impl<'span, 'history, 'ecx, 'mir, 'tcx> DiagnosticCx<'span, 'history, 'ecx, 'mir
                         None
                     }
                 })
+            }).or_else(|| {
+                // If we didn't find a retag that created this tag, it might be the base tag of
+                // this allocation.
+                if self.history.base.0.tag() == tag {
+                    Some((
+                        format!("{:?} was created here, as a base tag for {:?}", tag, self.history.id),
+                        self.history.base.1.data()
+                    ))
+                } else {
+                    None
+                }
```

Comment:
While developing this, I tried the reproducer for the current VecDeque issue:
```rust
use std::collections::VecDeque;

fn main() {
    let mut tester: VecDeque<usize> = VecDeque::with_capacity(3);

    for i in 0..3 {
        tester.push_back(i);
    }

    let _: VecDeque<_> = tester.drain(1..2).collect();
}
```
Which before this PR reports no additional helps, because it fails to find the creation of the tag being used in the error. Because the tag being used is the base tag, which we never store anywhere.